### PR TITLE
EDU-1183 Make notifications dismissable

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-1183

### CODE REVIEW

File: src/components/dashboard/notifications-tab.yml [hunk 1]
```diff
@@ -1,13 +1,28 @@
 notificationsTab:
   info:
-    en: Here you can find your notifications.
-    de: Hier findest du deine Benachrichtigungen.
+    en: Here you can find your notifications about contibutions to your favorite documents, by your favorite users or to documents in rooms that you are a member of.
+    de: Hier findest du deine Benachrichtigungen über Beiträge an deine Lieblingsdokumente, von deine Lieblingsnutzern oder an Dokumente in Räume, in denen du Mitglied bist.
```
Comment (ahelmberger) [inline src/components/dashboard/notifications-tab.yml:4]: Hier findest du Benachrichtigungen über Beiträge zu deinen Lieblingsdokumenten, von deine Lieblingsnutzern, oder zu Dokumenten in Räumen, in denen du Mitglied bist.

Maybe we should also scratch the first "your" in the english version?

File: src/components/dashboard/notifications-tab.yml [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
+    de: Raum-Dokument wurde bearbeitet
+  publicCommentCreatedNotification:
+    en: New comment on public document
+    de: Neuer Kommentar am öffentliches Dokument
+  roomCommentCreatedNotification:
+    en: New comment on room document
+    de: Neuer Kommentar am Raum-Dokument
+  documentNotAvailable:
+    en: Document no longer available
+    de: Document no longer available
```
Comment (ahelmberger) [inline src/components/dashboard/notifications-tab.yml:22]: Dokument nicht mehr verfügbar

**Comment by ahelmberger** [comment]
Looks super nice, I love it (also the activities) :heart: 

There's one logical thing that I would improve: in my opinion, the "remove all" should only remove the notifications that the user currently sees on the UI, not what's currently in the DB, meaning that we should only use `deleteNotificationsByNotifiedUserIdAndNotificationIds` and would not need `deleteNotificationsByNotifiedUserId`. Otherwise users would miss notifications that were created during their page load and the time they clicked the "remove all" button. Watcha think?

File: src/components/dashboard/notifications-tab.yml (hunk 2 above)
Comment (adrianaluchian) [reply src/components/dashboard/notifications-tab.yml:22]: Actually I knew this one 🤦‍♀️ Thanks!

**Comment by adrianaluchian** [comment]
> Looks super nice, I love it (also the activities) ❤️
> 
> There's one logical thing that I would improve: in my opinion, the "remove all" should only remove the notifications that the user currently sees on the UI, not what's currently in the DB, meaning that we should only use `deleteNotificationsByNotifiedUserIdAndNotificationIds` and would not need `deleteNotificationsByNotifiedUserId`. Otherwise users would miss notifications that were created during their page load and the time they clicked the "remove all" button. Watcha think?

Thank you!! ☺️ 

And super good point about removing currently visible notifications! I was thinking we could use the timestamp of the "remove all" request and remove the notifications created before this date, but that's not enough either. So I will change to actually use the same endpoint as for notification group removal and just pass all the ids from the UI.

**Comment by ahelmberger** [comment]
> > Looks super nice, I love it (also the activities) heart
> > There's one logical thing that I would improve: in my opinion, the "remove all" should only remove the notifications that the user currently sees on the UI, not what's currently in the DB, meaning that we should only use `deleteNotificationsByNotifiedUserIdAndNotificationIds` and would not need `deleteNotificationsByNotifiedUserId`. Otherwise users would miss notifications that were created during their page load and the time they clicked the "remove all" button. Watcha think?
> 
> Thank you!! relaxed
> 
> And super good point about removing currently visible notifications! I was thinking we could use the timestamp of the "remove all" request and remove the notifications created before this date, but that's not enough either. So I will change to actually use the same endpoint as for notification group removal and just pass all the ids from the UI.
> 
> **Update:** Fixed (along with another new issue for the case of deleted documents). Thanks again for pointing out this use case, good one! +1

Poifect :+1:

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**